File: app/src/main/java/com/codemybrainsout/imageviewer/view/single/SingleModule.java
package com.codemybrainsout.imageviewer.view.single;

import android.content.Context;

import com.codemybrainsout.imageviewer.dagger.scope.ActivityScope;
import com.codemybrainsout.imageviewer.view.photo.PhotoAdapter;

import dagger.Module;
import dagger.Provides;

/**
 * Created by ahulr on 11-06-2017.
 */

@Module
public class SingleModule {

    private Context context;

    public SingleModule(Context context) {
        this.context = context;
    }

    @ActivityScope
    @Provides
    Context context() {
        return context;
    }

    @ActivityScope
    @Provides
    PhotoAdapter photoAdapter(Context context) {
        return new PhotoAdapter(context);
    }

}

File: app/src/main/java/com/codemybrainsout/imageviewer/view/single/SingleModule.kt
package com.codemybrainsout.imageviewer.view.single

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

/**
 * Created by ahulr on 11-06-2017.
 */
@Module
@InstallIn(ActivityComponent::class)
abstract class SingleModule {

    @Binds
    abstract fun bindActivity(activity: SingleCollectionActivity): SingleContract.View

    @Binds
    abstract fun bindPresenter(impl: SinglePresenter): SingleContract.Presenter

}